### PR TITLE
feat: reorganize tags

### LINKED ISSUE #40: Re-organize test file tags
The tags need to be re-organized as discussed [here](https://github.com/elixir-cloud-aai/tes-compliance-suite/pull/29#discussion_r1133279601).

To Do
- [ ] Remove `name` field from test files and test schema. Use the test file/path name as the unique tag for each test.
- [ ] Remove all tags for individual tests and use `schema_validation_only` tag instead of the `logical` tag.
- [ ] Allow only letters, digits and underscores in tag names. The current --tag CLI arg needs to be repeated for multiple tags. It is also wrapped around quotes to support spaces in tag names. Use a more user-friendly syntax such as `--tag tag1,tag2,tag3`

### PR DESCRIPTION
**Pull request by LakiG**
Fixes #40

The following changes were implemented:
- Regex check on CLI args `--include-tags` and `--exclude-tags`
- Mandatory input for CLI args `--server` and `--version`
- Removal of `name` from tests
- Update tags for each test:
Tests that solely perform schema checks now have the tag `schema_validation_only`. To run only these tests, this command should be used `report --server SERVER --version VERSION --include-tags schema_validation_only`. To specifically exclude these schema validation tests and run the remaining tests, the exclusion feature can be utilized: `report --server SERVER --version VERSION --exclude-tags schema_validation_only`
- Condition to run all the tests:
If `--include-tags` is not provided, all tests will be executed by default (subject to exclusion criteria if `--exclude-tags` is provided). Example: `report --server SERVER --version VERSION` will run all available tests.

### CODE REVIEW

File: compliance_suite/cli.py [hunk 1]
```diff
@@ -51,15 +77,11 @@ def report(server: str,
         uptime (int): The local server duration in seconds. Default - 3600 seconds
     """
 
-    if server is None:
-        raise Exception("No server provided. Please provide a server URL.")
-
-    if version is None:
-        raise Exception("No version provided. Please provide a version.")
```
Comment (LakiG) [inline compliance_suite/cli.py:58]: Using `click`'s `prompt` ensures that a value is provided in the option. So, we can remove these checks.

File: compliance_suite/utils/test_utils.py [hunk 2]
```diff
@@ -23,7 +23,9 @@ def tag_matcher(
         present in the yaml_tags. Otherwise, False.
     """
 
+    # If include_tags is empty, all tests are considered included
+
     return (
         not any(exclude_tag in yaml_tags for exclude_tag in exclude_tags)
-        and any(include_tag in yaml_tags for include_tag in include_tags)
+        and (not include_tags or any(include_tag in yaml_tags for include_tag in include_tags))
```
Comment (LakiG) [inline compliance_suite/utils/test_utils.py:30]: Contrary to what we discussed, I used this approach where we assume all tests are included by default. The elimination will be decided depending on the `--exclude-tags`.
This also allows us to handle tests which have no tags (otherwise the check was failing for empty `yaml_tags`)

Comment (uniqueg) [reply compliance_suite/utils/test_utils.py:30]: I'm not sure how this differs from what we discussed. It's probably fine, but just to make sure we are on the same page, could you give one or more examples for this behavior? I have very little time and it's not trivial to read that code and fully understand the implications of it.

**Review by uniqueg — COMMENTED** [review]
Please see my comment on the open conversation. Actually, I think it would make sense to add the examples to the PR description (and actually _add_ a description).

File: compliance_suite/utils/test_utils.py (hunk 2 above)
Comment (LakiG) [reply compliance_suite/utils/test_utils.py:30]: We initially considered allowing a test to run if neither `--include-tags` nor `--exclude-tags` were provided. In that case, the code change would have been:
```
if include_tags == [] and exclude_tags ==[]:
  return True
```
This would have worked perfectly, but we also updated the tags of each test. Some tests, specifically functional tests, no longer have any tags assigned to them, resulting in an empty `yaml_tags` list. Therefore, both conditions in `return` statement evaluated to `False`, leading to the simplified expression `return (not False and False)`, i.e. `return False`. Consequently, the functional tests were never executed. Handling this situation would have required unnecessary additional changes.

Therefore, I modified the code to assume a test is included by default if `--include-tags` is empty, which actually makes sense. We can then proceed to check the exclusion condition.

In summary, rather than touching `--exclude-tags`, I updated the condition based solely on `--include-tags`. The exclusion mechanism remains the same: to exclude a test, `--exclude-tags` must be provided and matched. For inclusion, if `--include-tags` is provided, it needs to match the test tags. Otherwise, the test is included by default.

Comment (LakiG) [reply compliance_suite/utils/test_utils.py:30]: Earlier, the execution of all tests was handled by the tag `all`. After its removal, we needed a way to execute all tests and thus the above way was implemented.

Comment (uniqueg) [reply compliance_suite/utils/test_utils.py:30]: I'm sorry, but I still don't get it 🙈😅 If I want to run all test, do I need to specify something or not?

Comment (LakiG) [reply compliance_suite/utils/test_utils.py:30]: No problem :smile:.  No, you dont need to specify anything to run all tests.

> Condition to run all the tests:
> If --include-tags is not provided, all tests will be executed by default (subject to exclusion criteria if --exclude-tags is provided). Example: report --server SERVER --version VERSION will run all available tests

**Review by uniqueg — APPROVED** [review]
